cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -327,7 +327,7 @@ int VpiSignalObjHdl::set_signal_value(long value)
 {
     s_vpi_value value_s;
 
-    value_s.value.integer = value;
+    value_s.value.integer = (int)value;
```

Comment:
This looks like a bug to me - although one that was already here. Worth at least calling out with a TODO and a github issue, if you don't want to think about fixing it now.

Reply:
The correct solution would be making set_signal_value take an int instead of a long. I've opened https://github.com/cocotb/cocotb/issues/1202 to take care of that.